### PR TITLE
fix(coderd/database): improve perf of `GetTemplateInsightsByInterval`

### PR DESCRIPTION
**Pull request by mafredri**
New: https://explain.dalibo.com/plan/ebf9692d3426f3bd
Old: https://explain.dalibo.com/plan/62eg2ca495g6f2gg

3s -> 61ms.

Refs #12122

### CODE REVIEW

**Comment by mafredri** [comment]
* **#12773** <a href="https://app.graphite.dev/github/pr/coder/coder/12773?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mafredri and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: coderd/database/queries/insights.sql [hunk 1]
```diff
@@ -389,14 +389,17 @@ WITH
 	ts AS (
 		SELECT
 			d::timestamptz AS from_,
-			CASE
-				WHEN (d::timestamptz + (@interval_days::int || ' day')::interval) <= @end_time::timestamptz
-				THEN (d::timestamptz + (@interval_days::int || ' day')::interval)
-				ELSE @end_time::timestamptz
-			END AS to_
+			LEAST(
```
Comment (mafredri) [inline coderd/database/queries/insights.sql:392]: `LEAST` is more performant than case.

File: coderd/database/queries/insights.sql [hunk 2]
```diff
@@ -410,6 +413,7 @@ LEFT JOIN
 	template_usage_stats AS tus
 ON
 	tus.start_time >= ts.from_
+	AND tus.start_time < ts.to_ -- End time exclusion criteria optimization for index.
```
Comment (mafredri) [inline coderd/database/queries/insights.sql:416]: Main performance optimization.

Comment (mtojek) [reply coderd/database/queries/insights.sql:416]: Nice spotting 👍 👍

**Review by mtojek — APPROVED** [review]

**Pull request merged**